beets/util/artresizer.py

```diff
@@ -51,30 +61,100 @@ class LocalBackendNotAvailableError(Exception):
     pass
 
 
-_NOT_AVAILABLE = object()
+# Singleton pattern that the typechecker understands:
+# https://peps.python.org/pep-0484/#support-for-singleton-types-in-unions
+class NotAvailable(Enum):
+    token = 0
+
+
+_NOT_AVAILABLE = NotAvailable.token
 
 
-class LocalBackend:
+class LocalBackend(ABC):
+    NAME: ClassVar[str]
+
+    @classmethod
+    @abstractmethod
+    def version(cls) -> Any:
+        pass
+
     @classmethod
-    def available(cls):
+    def available(cls) -> bool:
         try:
             cls.version()
             return True
         except LocalBackendNotAvailableError:
             return False
 
+    @abstractmethod
+    def resize(
```

Comment:
Would it be possible to get some docstrings for the abstract methods?  I'm not too familiar with the artresizer, a sentence on each each of them would be highly appreciated.

Reply:
Done.